src/src5 implementation.md

```diff
@@ -0,0 +1,113 @@
+## SRC-5: A comprhensive Overview
+
+SRC-5 is a standard for smart contract interface introspection in Starknet, inspired by the Ethereum ERC-165 standard. It provides a method for contracts to publish and detect the interfaces they implement, ensuring standardized interaction. It provides a method for contracts to publish and detect the interfaces they implement, ensuring standardized interaction.  You can find more information and details refer to the [SRC-5 specification](https://github.com/starknet-io/SNIPs/blob/main/SNIPS/snip-5.md#simple-summary) and the [Ethereum ERC-165 standard](https://eips.ethereum.org/EIPS/eip-165).
+
+
+
+
+### SRC-5 offers a standardized method to:
+
+* Identify interfaces.
+* Publish the interfaces a contract implements.
+* Detect if a contract implements any given interface. (including SRC-5)
+
+
+### Interface Definition
+An interface is a set of function signatures with specific type parameters, represented as traits. These traits are meant to be implemented externally by compliant contracts.
+
+Example:
+
+    ```rust 
+    trait IMyContract {
+    fn foo(some: u256) -> felt252;
+    }
+    ```
+With Cairo 2.0, generic traits can represent a set of interfaces:
+
+    ```rust
+    #[starknet::interface]
+    trait IMyContract<TContractState, TNumber> {
+    fn foo(self: @TContractState, some: TNumber) -> felt252;
+    }
+    ```
+#### Extended Function Selector
+The function selector in Starknet is the starknet_keccak hash of the function name. The extended function selector, for SRC-5, is the starknet_keccak hash of the function signature:
+
+fn_name(param1_type,param2_type,...)->output_type
+
+For example,for a function with zero parameters and no return value:
+
+
+    fn_name()
+
+Special Types:
+```
+* Tuples: (elem1_type,elem2_type,...)
+* Structs: (field1_type,field2_type,...)
+* Enums: E(variant1_type,variant2_type,...)
+```
+
+#### Interface Identification
+An interface identifier is the XOR of all extended function selectors in the interface.
+
+Example (Python code):
+
+
+    from starkware.starknet.public.abi import starknet_keccak
+
+    signatures = [
+    'supports_interface(felt252)->E((),())',
+    'is_valid_signature(felt252,Array<felt252>)->E((),())',
+    '__execute__(Array<(ContractAddress,felt252,Array<felt252>)>)->Array<(@Array<felt252>)>',
+    '__validate__(Array<(ContractAddress,felt252,Array<felt252>)>)->felt252',
+    '__validate_declare__(felt252)->felt252'
+    ]
+
+    def compute_interface_id():
+    interface_id = 0x0
+    for sig in signatures:
+        function_id = starknet_keccak(sig.encode())
+        interface_id ^= function_id
+    print('IAccount ID:', hex(interface_id))
+
+    compute_interface_id()
```

Comment:
Also, it might be a good idea to move this code snippet inside the listing directory in a python file